mauth_client/middlewares/asgi.py

```diff
@@ -0,0 +1,115 @@
+import json
+import logging
+
+from asgiref.typing import (
+    ASGI3Application,
+    ASGIReceiveCallable,
+    ASGIReceiveEvent,
+    ASGISendCallable,
+    Scope,
+)
+from typing import List, Tuple
+
+from mauth_client.authenticator import LocalAuthenticator
+from mauth_client.config import Config
+from mauth_client.consts import (
+    ENV_APP_UUID,
+    ENV_AUTHENTIC,
+    ENV_PROTOCOL_VERSION,
+)
+from mauth_client.signable import RequestSignable
+from mauth_client.signed import Signed
+from mauth_client.utils import decode
+
+logger = logging.getLogger("mauth_asgi")
+
+
+class MAuthASGIMiddleware:
+    def __init__(self, app: ASGI3Application, exempt: set = set()) -> None:
+        self._validate_configs()
+        self.app = app
+        self.exempt = exempt
+
+    async def __call__(
+        self, scope: Scope, receive: ASGIReceiveCallable, send: ASGISendCallable
+    ) -> None:
+        path = scope["path"]
+
+        if path in self.exempt:
+            return await self.app(scope, receive, send)
```

Comment:
Should we also skip processing within this middleware if `"http" not in scope["type"]`? I don't think we currently support WebSockets in MAuth do we?

Reply:
interesting point.. theoretically we should be able to use MAuth for websocket as the initial websocket req is just a http GET.  i think getting that working is beyond the scope of this initial PR, but agree that we should examine the type so as not to block websocket reqs.  will add that exception and a test.

Reply:
[0e9c0b2](https://github.com/mdsol/mauth-client-python/pull/34/commits/0e9c0b243d39759f2f023cb2c94ff86bf759840c)